src/Diagnostics.Logger/ApiMetricsLogger.cs

```diff
@@ -49,6 +49,16 @@ public void LogRuntimeHostAPIMetrics(HttpContext context)
 
             endTime = DateTime.UtcNow;
             latencyInMs = Convert.ToInt64((endTime - startTime).TotalMilliseconds);
+            var headers = context.Request.Headers;
+            var headersContent = "Headers received: ";
+            foreach (var header in headers)
+            {
+                if (header.Key.StartsWith("x-ms", StringComparison.OrdinalIgnoreCase))
+                {
+                    headersContent += $"{header.Key} - ";
+                    headersContent += string.IsNullOrWhiteSpace(header.Value.FirstOrDefault()) ? string.Empty : "****";
```

Comment:
QQ - Have you considered the ease of writing an alert if we don't get x-ms-client-id header passed? 
I think the format in which we are writing this log message can be improved.

Should you log them as a JSON string like this which might make it easier to parse and extensible for future?
{
"requestHeaders" : [{"name": "x-ms-client-id", "value": "*****"}, {}, ...]
"someOtherFieldInFuture": "some Value"
}

Reply:
I agree, I will make changes to log it a JSON string. 
To check if RunTimeHost received the x-ms-client-id header, we can check against the incoming Request.Headers. I was planning to follow up (while PR is in review) regarding how we configure alerts.

Reply:
Changed to log headers as JSON string.
![image](https://user-images.githubusercontent.com/46545195/55015543-4a3c2100-4faa-11e9-941a-89b082c72d9f.png)